pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -411,6 +422,7 @@ func (nc *NodeController) monitorNodeStatus() error {
 				}
 				if _, err := instances.ExternalID(node.Name); err != nil && err == cloudprovider.InstanceNotFound {
 					glog.Infof("Deleting node (no longer present in cloud provider): %s", node.Name)
+					nc.recordNodeEvent(node.Name, fmt.Sprintf("Deleting Node %v because it's not present according to cloud provider", node.Name))
```

Comment:
am i right in assumin we don't delete the pods in this case (and if so, why not)?